ci: add Cygwin build workflow

ritesh006 opened the pull request:
<!--- Make sure to read CONTRIBUTING.md -->
<!--- It mentions the rules to follow and helpful tools -->

<!-- For completed items, change [ ] to [x]. -->
- [ ] I ran valgrind when using my new feature

### How to test the functionality
* step 1
* step 2

jubalh commented:
It's just a way to signal whether your pull request is still work in progress or ready for review by maintainers.

https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request

jubalh commented on .github/workflows/cygwin-build.yml:
```diff
@@ -0,0 +1,91 @@
+name: Cygwin build
```
I maybe would just call it `Cygwin` like the others are also called `Arch` etc

jubalh commented on .github/workflows/cygwin-build.yml:
```diff
@@ -0,0 +1,91 @@
+          cd "$GITHUB_WORKSPACE"
+          make -j2
+
+      # Uncomment when tests are reliable on Cygwin
```
What is the plan with this?

jubalh reviewed (changes requested):
Now you need to clean up your history.
The commit [Merge branch 'master' into ci/cygwin-build](https://github.com/profanity-im/profanity/pull/2066/commits/f8d1ae0741af59c47c17b7dd474923204fd37f1a) is not needed and just clutters the history. The other 3 commits can be squashed into one.

ritesh006 commented:
Hi @jubalh due to family loss I am in hometown as soon as I reach to my place would like to work start again. Thank You

ritesh006 commented:
> Now you need to clean up your history. The commit [Merge branch 'master' into ci/cygwin-build](https://github.com/profanity-im/profanity/pull/2066/commits/f8d1ae0741af59c47c17b7dd474923204fd37f1a) is not needed and just clutters the history. The other 3 commits can be squashed into one.

I cleaned the branch history and squashed the fixes into a single commit.

jubalh commented:
Sounds good!

Regarding:

> enable make check later if tests are stable on Cygwin,

The tests do not succeed there currently?

ritesh006 commented:
I added a Cygwin workflow that installs required Cygwin packages, builds libstrophe, and runs autoreconf -fi, ./configure, and make.
I haven’t enabled make check as a blocking step yet — I can add a non-blocking test run that uploads logs so we can triage failures, then enable tests once they’re stable.

ritesh006 commented:
> Sounds good!
> 
> Regarding:
> 
> > enable make check later if tests are stable on Cygwin,
> 
> The tests do not succeed there currently?

could you please help me here bacuse its failing at linux stage not at Cygwin I thing Cygwin not even started yet

jubalh commented:
I'm not sure why it doesn't use the new workflow file. Can you try cleaning up git history into 1 commit and force push?

sjaeckel commented:
BTW there's https://github.com/nektos/act to run GH actions locally while working on them. Not sure how/if it works for Windows runners, but it works great for Linux CI jobs.

jubalh commented on .github/workflows/cygwin-build.yml:
```diff
@@ -0,0 +1,120 @@
+          cd "$GITHUB_WORKSPACE"
+          make -j2
+
+      # Optional: non-blocking tests + upload logs for inspection (recommended while tests are unstable)
```
Everything looks great :)
But I still don't understand what you mean with the tests being unstable. Or what the problem with them is. Is there one?

ritesh006 replied:
If everything looks fine, can we merge these changes? After that, we can handle the remaining parts in the next commit

jubalh replied:
This comment will be on master and even though people in the past often promised to change something in the future, sometimes they didn't do it. So I would like to understand this comment first. Or removing the outcommented part and just adding the whole thing once its working.

ritesh006 replied:
Could you please provide me with the definition of done? That will help me understand your expectations better. Since no issue was created for this, I just want to clarify

And I assure you I will give my best

jubalh replied:
The comment above mentions that the tests are "unstable". I don't know what you mean by that.

ritesh006 replied:
I uncommented the test step and it’s passing now. I kept the note so if tests fail in the future we remember why. If you want, I can remove the comment.

sjaeckel commented:
> please take a look

I took a look now. Could you please address all the comments?

jubalh commented:
@ritesh006 thanks for the update. Currently the CI test fails though.

ritesh006 commented:
> @ritesh006 thanks for the update. Currently the CI test fails though.

@jubalh CI success please review.

ritesh006 commented:
> Unfortunately now there is again a merge commit. Like already mentioned in [#2066 (review)](https://github.com/profanity-im/profanity/pull/2066#pullrequestreview-3223383080) we dont like them :) A rebase on master would have been more appropriate. Please get rid of that second commit.

Please check now is everything ok.

ritesh006 commented on .github/workflows/cygwin-build.yml:
```diff
@@ -0,0 +1,120 @@
+          cd "$GITHUB_WORKSPACE"
+          make -j2
+
+      # Optional: non-blocking tests + upload logs for inspection (recommended while tests are unstable)
```
done

ritesh006 commented:
I addressed all remaining review comments:

- simplified the workflow and removed debug-only steps,

- removed redundant shell overrides,

- added a short comment explaining the Windows/Cygwin wrapper and bootstrap.sh normalization,

- removed test-related logic so CI no longer fails.

The PR now only ensures Profanity builds on Cygwin. Test support can be added in a follow-up PR.

Thanks for the review!

jubalh commented:
I didn't forget about it. I was messaging jaeckel today asking him whether he is okay with your comments. Once he replied we can continue.
Because for example: https://github.com/profanity-im/profanity/pull/2066#discussion_r2428104145 didnt get a comment.
